### PR TITLE
Cleaning up "user" frontend path.

### PR DESCRIPTION
**Pull request by mikzat**
There should be no functional changes from any of this. 

See description of each individual commit for details.

### CODE REVIEW

File: frontend/rolecall/src/app/user/user-editor.component.spec.ts [hunk 1]
```diff
@@ -1,40 +1,30 @@
...
-      declarations: [UserEditor, EmptyStringIfUndefinedPipe],
-      providers: [
-        {
-          provide: ActivatedRoute,
-          useValue: activatedRouteStub
-        }
-      ]
-    })
-      .compileComponents();
+          declarations: [UserEditor, EmptyStringIfUndefinedPipe],
```
Comment (yhedholm) [inline frontend/rolecall/src/app/user/user-editor.component.spec.ts:15]: You did more than pretty this one up.

File: frontend/rolecall/src/app/user/user-editor.component.ts [hunk 2]
```diff
@@ -1,121 +1,197 @@
...
       }
     }
   }
 
-  setCurrentUser(user: User, fromInputChange?: boolean, shouldSetLastUser?: boolean) {
+  setCurrentUser({user, fromInputChange, shouldSetLastUser}: {
+    user: User | undefined,
+    fromInputChange?: boolean,
+    shouldSetLastUser?: boolean,
+  }) {
```
Comment (yhedholm) [inline frontend/rolecall/src/app/user/user-editor.component.ts:157]: Is this to force the caller to specify parameters?

File: frontend/rolecall/src/app/user/user-editor.component.ts [hunk 3]
```diff
@@ -1,121 +1,197 @@
...
+      private readonly route: ActivatedRoute,
+      private readonly userAPI: UserApi,
+      private readonly location: Location) {
+  }
 
   ngOnInit(): void {
-    let uuid = this.route.snapshot.params.uuid;
-    if (!isNullOrUndefined(uuid)) {
+    const uuid: string = this.route.snapshot.params.uuid;
+    if (uuid) {
```
Comment (yhedholm) [inline frontend/rolecall/src/app/user/user-editor.component.ts:90]: You don't always specify a type when assigning a const. Why here?

**Review by yhedholm — APPROVED** [review]
Thank you! I look forward to working with this new and improved file!

File: frontend/rolecall/src/app/user/user-editor.component.ts (hunk 3 above)
Comment (mikzat) [reply frontend/rolecall/src/app/user/user-editor.component.ts:90]: Good eye. The reason for changing it here is that `params` has the type `"[key: string]: any"` so the value of `uuid` evaluates to `"any"`. Narrowing the type as early as possible allows the rest of the code to know that it's actually a string for proper type checking.

File: frontend/rolecall/src/app/user/user-editor.component.ts (hunk 2 above)
Comment (mikzat) [reply frontend/rolecall/src/app/user/user-editor.component.ts:157]: Correct, this allows callers to be more readable as it's clear what parameters are being set. It also makes the arguments position-agnostic even though they are optional.

**Pull request merged**